src/pymodaq/daq_viewer/daq_viewer_main.py

```diff
@@ -1843,31 +1980,58 @@ def process_tcpip_cmds(self, status):
 
 
 class DAQ_Detector(QObject):
-    """
-        ========================= ==========================
-        **Attributes**              **Type**
-        *status_sig*                instance of pyqt Signal
-        *data_detector_sig*         instance of pyqt Signal
-        *data_detector_temp_sig*    instance of pyqt Signal
-
-        *waiting_for_data*          boolean
-        *controller*                ???
-        *detector_name*             string
-        *detector*                  ???
-        *controller_adress*         ???
-        *grab_state*                boolean
-        *single_grab*               boolean
-        *x_axis*                    1D numpy array
-        *y_axis*                    1D numpy array
-        *datas*                     dictionnary
-        *ind_average*               int
-        *Naverage*                  int
-        *average_done*              boolean
-        *hardware_averaging*        boolean
-        *show_averaging*            boolean
-        *wait_time*                 int
-        *DAQ_type*                  string
-        ========================= ==========================
+    """This class makes the bridge between the DAQ_Viewer class (UI) and the pymodaq plugin that corresponds to the
+    currently plugged detector.
+
+    It is intended to be put in a thread.
+
+    Signals
+    -------
+    data_detector_sig : pyqt signal
+        This signal hands over the acquisition from DAQ_Detector to DAQ_Viewer.
+        Emitted by data_ready.
+        Connected to DAQ_Viewer.show_data.
+
+    Attributes
+    ----------
+    detector : a viewer plugin object (depends on the context, e.g. DAQ_1DViewer_Mock object).
+        The pymodaq viewer plugin that corresponds to the detector that is currently plugged.
+    wait_time : int
+        This corresponds to the "Wait time (ms)" in the parameters of the UI of the DAQ_Viewer (Main Settings).
+        Extra waiting time (in ms) in the pymodaq process to slow down the acquisition loop. Notice that it is not only
+        a matter of displaying on the UI. Of course the displaying on the UI cannot be faster than the loop timing, but
+        one could refresh the UI displaying slower than the actual speed of the acquisition. It actually reduces the
+        number of acquisition per second and thus gives a lower limit to the number of acquisition you can save per
+        second (see grab_data method).
+        !!!Not to be confused with DAQ_Viewer.wait_time!!!
+    grab_state : bool
+        True if the viewer is grabbing, False otherwise.
+    single_grab : bool
+        True if the last acquisition order from the user was a snap (single grab), False otherwise.
+    Naverage : int
+        ??
```

Comment:
integer to track the total number of acquisition to be averaged before displaying. this averaging is done in the DAQ_Detector if the plugin has not the self.hardware_averaging set to True else it should be done inside the plugin. For instance, it could be smarter to do the averaging inside a camera plugin when one ofen use a buffer and where it is faster to do so than exporting (and copying) all data and then averaging